### PR TITLE
Simple Noop for LOCK and UNLOCK

### PR DESCRIPTION
**Pull request by saifalharthi**
This addresses https://github.com/vitessio/vitess/issues/5402
Functionality of locking a table is not implemented

### CODE REVIEW

File: go/vt/sqlparser/analyzer.go [hunk 1]
```diff
@@ -110,6 +112,10 @@ func Preview(sql string) StatementType {
 		return StmtUse
 	case "analyze", "describe", "desc", "explain", "repair", "optimize":
 		return StmtOther
+	case "lock":
+		return StmtLock
+	case "unlock":
+		return StmtLock
```
Comment (deepthi) [inline go/vt/sqlparser/analyzer.go]: You mean StmtUnlock?

**Comment by deepthi** [comment]
If this is for mysqldump, we know this courtesy @aquarapid:
```
seems mysqldump with
  --skip-lock-tables --set-gtid-purged=OFF --add-locks=false
gets rid of all the objectionable stuff, including the session variables for community 8.0.x at least
and the result is importable via vtgate
```

**Comment by saifalharthi** [comment]
@deepthi I had a discussion with @morgo and @aquarapid and we just need to just pass it through at this time.

**Comment by saifalharthi** [comment]
@deepthi I saw that mysqldump flags yesterday. This just to cover is those flags (at least the ones concerning lock tables) were not used.

File: go/vt/sqlparser/analyzer.go (hunk 1 above)
Comment (saifalharthi) [reply go/vt/sqlparser/analyzer.go]: Correct!, this was fixed

**Comment by morgo** [comment]
I agree with @deepthi. If we do decide to make some unsafe features be noops, we should hide them behind a configuration setting such as `allow_unsafe_features_for_compatibility`. The default needs to be `FALSE`.

**Comment by sougou** [comment]
Agree with @morgo and @deepthi. Let's close this and rethink the right way to do this.

**Pull request closed without merging**